File: algorithms/Recursion/Sum/sum.go
package main

import (
	"fmt"
)

func sum(numList []int) int {
	if len(numList) == 1 {
		return numList[0]
	} else {
		return numList[0] + sum(numList[1:])
	}
}

func main() {
	numList := []int{1, 2, 3}
	fmt.Println(sum(numList))
}

File: algorithms/Recursion/Sum/sum.js
function sum(numList) {
    if (numList.length == 1)
        return numList[0]
    else {
        var num = numList[0]
        numList.splice(0, 1)
        return num + sum(numList)
    }
}

console.log(sum([1, 2, 3]))